Readme.md
# User Manual
## Quick Start/Stop:
- Use the green "Run" button to begin the program
- Use the grey "Stop" button to stop the program. Contacts will not save in this way.
## Main Menu Navigation:
- Enter '1' to view current contacts. If nothing but a list comes up, you have not entered any contacts and will need to return to the main menu. Otherwise, you can select to return back to the original menu before this screen. Entering '2' will delete all contacts. If entering '1', please refer to "Editing a Contact" 
- Enter '2' to manually add a new contact and fill out all the information.
- Enter '3' to save all your contacts onto the "SAVED_CONTACTS.txt" file. As well, this will close the program and permanently delete any contacts that were deleted during the run of the program.
- Enter '4' to quickly create 3 new pre-selected contacts. This is used for testing. This will only work once for each run of the program. It is not recommended to repeatedly use this, as it will continue adding the same contacts.
## Editing a Contact:
- When executed, a number will show up above each of your contacts. Enter the number above to edit the contact below.
- Afterwards, entering '1', '2', '3' or '4' will allow you to change that specific section of the contact. '5' will delete that contact entirely.
## What this program does different:
- This program stores and saves information intended to be associated with a person, including their name, email, phone number and other information wished to be stored by the user. It is quick, clear and concise while using a python coding program.
## Targeted Audience:
- The intended audience is those who want an efficient, easy to use and understand contact organizer using a python coding program.

Changelog.md
# Change Log
## 1/18/22 (beta v0.2.1)
- Lots of debugging 
- Added "Alpha_Beta_Testing.md"
- Added documentation of alpha and beta testing and requirement. 
## 1/17/22(beta v0.2.0)
- Reduced code lines for efficiency
- UI was improved
## 1/13/22 (alpha v0.1.4) 
- Added save contact function 
- Need to fix UI 
## 1/12/22 (alpha v0.1.3)
- Added descriptions and Readme file
## 1/4/22 (alpha v0.1.2)
- Added ability to reset and delete contacts
## 12/13/21 (alpha v0.1.1)
- Contact display now scales with the amount of contacts
## 12/9/21 (alpha v0.1.0)
- Added example contacts option
- Now able to continuously add more contacts and get proper feedback 
- Code is now usable
## 12/7/21 (v0.0.3)
- Call for inputted information from the menu
## 12/2/21 (v0.0.2)
- Added implementation of contacts and feedback
- 
## 11/25/21 (v0.0.1)
- Created files and changelog


ADD.py
# Course: CS 30
# Period: 3
# File created 11/25/21
# Last edited 01/19/22
# Name: Ethan Behl
# Description: File that carries functions and beginning variables
# Imports
try:
    import CONTACTS
    from os import system, name
    import sys
except ImportError:
    print("Import error on MAINFILE")
    sys.exit()


def clear():
    """Function that clears the user's screen"""
    if name == 'nt':
        system('cls')
    else:
        system('clear')


# Contact lists
contact_1 = []
contact_2 = []
contact_3 = []
contact_4 = []
contact_5 = []
contact_6 = []
contact_7 = []
contact_8 = []
contact_9 = []
contact_10 = []
# Starting variables that determine code functions
current_contact = contact_1
full_name = None
phone_number = None
email = None
con_num = 1
skip = 0


def CHECK_CONTACT():
    """Adjusts the current contact to the appropriate list"""
    global current_contact
    if CONTACTS.contact == 1:
        current_contact = contact_1
    elif CONTACTS.contact == 2:
        current_contact = contact_2
    elif CONTACTS.contact == 3:
        current_contact = contact_3
    elif CONTACTS.contact == 4:
        current_contact = contact_4
    elif CONTACTS.contact == 5:
        current_contact = contact_5
    elif CONTACTS.contact == 6:
        current_contact = contact_6
    elif CONTACTS.contact == 7:
        current_contact = contact_7
    elif CONTACTS.contact == 8:
        current_contact = contact_8
    elif CONTACTS.contact == 9:
        current_contact = contact_9
    elif CONTACTS.contact == 10:
        current_contact = contact_10


def ADD_CONTACT():
    """The function that adds contacts (Info into a list)"""
    CHECK_CONTACT()
    first_name_in = input("What is the first name of your contact?")
    last_name_in = input("What is the last name of your contact?")
    if first_name_in == "":
        if last_name_in == "":
            current_contact.append("No name entered.")
    else:
        full_name = (str(first_name_in) + " " + str(last_name_in))
        current_contact.append(full_name)
    phone_in = input("""
Would you like to enter a phone number?
[1] Yes
[2] No
""")
    if phone_in == "1":
        # Phone number information input
        phone_area_in = input("What is the area code of" +
        "their phone number?")
        phone_3diget_in = input("What is the 3 diget" +
        "set of their phone number?")
        phone_4digit_in = input("What is the 4 digit" +
        "set of their phone number?")
        phone_number = ("(" + str(phone_area_in) + ") " +
        str(phone_3diget_in) + "-" + str(phone_4digit_in))
        current_contact.append(phone_number)
    else:
        # Stand in when phone number not added
        print("\nNo phone number entered.")
        current_contact.append("No phone number entered.")
    email_in = input("""
Would you like to enter an email?
[1] Yes
[2] No
""")
    if email_in == "1":
        # Email information input
        email = input("What is their email?")
        current_contact.append(email)
    else:
        # Stand in when email not added
        print("\nNo email entered.")
        current_contact.append("No email entered.")
    desc_in = input("""
Would you like to enter any additional information?
[1] Yes
[2] No
""")
    if desc_in == "1":
        # Description information input
        desc = input("What information would you like to add?")
        current_contact.append(desc)
    else:
        # Stand in when description not added
        print("No description entered.")
        current_contact.append("No description entered.")
    CONTACTS.contact += 1


# contact goes down, num goes up
def INFO():
    """Prints out the contact information"""
    global con_num
    CONTACTS.contact -= 1
    CHECK_CONTACT()
    con_num = 0
    only_alpha = ""
    try:
        for char in current_contact[0]:
            if char.isalpha():
                only_alpha += char
    except IndexError:
        None
    while CONTACTS.contact >= 1:
        # Prints out contacts for display
        CHECK_CONTACT()
        print("\n" + current_contact[0])
        print(" > " + current_contact[1])
        print(" > " + current_contact[2])
        print(" > " + current_contact[3])
        con_num += 1
        CONTACTS.contact -= 1
    CONTACTS.contact = con_num
    CONTACTS.contact += 1
    CHECK_CONTACT()


def EXAMPLE_CONTACTS():
    """ Applies the example contacts"""
    CHECK_CONTACT()
    current_contact.append("Mike Untz")
    current_contact.append("(306) 241-6968")
    current_contact.append("sussybaka@hotmail.com")
    current_contact.append("No description entered.")
    CONTACTS.contact += 1
    CHECK_CONTACT()
    current_contact.append("Ashoosh Poonda")
    current_contact.append("No phone number entered.")
    current_contact.append("pandalad@icloud.com")
    current_contact.append("No description entered.")
    CONTACTS.contact += 1
    CHECK_CONTACT()
    current_contact.append("Biga Sole")
    current_contact.append("(639) 555-4336")
    current_contact.append("No email entered.")
    current_contact.append("No description entered.")
    CONTACTS.contact += 1
    CHECK_CONTACT()


def MODIFY_CONTACT():
    """Controls the modifications of all contacts"""
    global con_num
    global skip
    clear()
    print("Which contact would you like to modify?")
    CONTACTS.contact -= 1
    CHECK_CONTACT()
    con_num = 0
    while CONTACTS.contact >= 1:
        # Loop that prints out contacts to user for modification
        CHECK_CONTACT()
        print((str(CONTACTS.contact)) + "\n" + (str(current_contact[0])) +
        "\n > " + (str(current_contact[1])) +
        "\n > " + (str(current_contact[2])) +
        "\n > " + (str(current_contact[3])))
        con_num += 1
        CONTACTS.contact -= 1
    mod_contact_in = input(" ")
    try: 
        CONTACTS.contact = int(mod_contact_in)
    except ValueError:
        skip = 1
    if skip == 1:
        skip = 0
    elif con_num < int(mod_contact_in):
        None
    else:
        CHECK_CONTACT()
        print(str(current_contact))
        # Mod menu
        mod_info_in = input("""
What information do you want to modify?
[1] Name
[2] Phone Number
[3] Email
[4] Description
[5] Delete Contact
""")
        if mod_info_in == "1":
            # Replace the current contact's name
            first_name_in = input("What is the first name of your contact?")
            last_name_in = input("What is the last name of your contact?")
            full_name = (str(first_name_in) + " " + str(last_name_in))
            current_contact[0] = full_name
        elif mod_info_in == "2":
            # Replace the current contact's phone number
            phone_area_in = input("What is the area code of" +
            "their phone number?")
            phone_3diget_in = input("What is the 3 diget" +
            "set of their phone number?")
            phone_4digit_in = input("What is the 4 digit" +
            "set of their phone number?")
            phone_number = ("(" + str(phone_area_in) + ") " +
            str(phone_3diget_in) + "-" + str(phone_4digit_in))
            current_contact[1] = phone_number
        elif mod_info_in == "3":
            # Replace the current contact's email
            email = input("What is their email?")
            current_contact[2] = email
        elif mod_info_in == "4":
            # Replace the current contact's description
            desc = input("What information would you like to add?")
            current_contact[3] = desc
        elif mod_info_in == "5":
            # Delete the current contact, replaced with place holders
            current_contact.clear()
            current_contact.append("CONTACT DELETED")
            current_contact.append("Reload application to delete this.")
            current_contact.append(" ")
            current_contact.append(" ")
        CONTACTS.contact = con_num
        CONTACTS.contact += 1
        CHECK_CONTACT()


def RESET_CONTACTS():
    """Resets all contacts"""
    global current_contact
    CONTACTS.contact -= 1
    while CONTACTS.contact >= 1:
        CHECK_CONTACT()
        current_contact.clear()
        CONTACTS.contact -= 1
    CONTACTS.contact += 1


MAINFILE.py
# Course: CS 30
# Period: 3
# File created 11/25/21
# Last edited 01/19/22
# Name: Ethan Behl
# Description: Main file for the Contact Organizer
# Imports
try:
    import ADD
    import CONTACTS
    import sys
except ImportError:
    print("Import error on MAINFILE")
    sys.exit()
# Starting variables that determine a couple code functions
example = True
MAIN = False
# Initiates the text files of the program for saved contacts
file = open("SAVED_CONTACTS.txt", "r")
list_of_lists = [(line.strip()).split() for line in file]
file.close()
print(list_of_lists)
ADD.CHECK_CONTACT()
# The following variables control the while loop for applying the saved contact
y = 1
x = 1
while y <= 9:
    try:
        ADD.CHECK_CONTACT()
        ADD.current_contact.append(str(list_of_lists[x]))
        x += 1
        ADD.current_contact.append(str(list_of_lists[x]))
        x += 1
        ADD.current_contact.append(str(list_of_lists[x]))
        x += 1
        ADD.current_contact.append(str(list_of_lists[x]))
        x += 1
        y += 1
        ADD.current_contact[0] = ADD.current_contact[0].replace("[", "")
        ADD.current_contact[0] = ADD.current_contact[0].replace("]", "")
        ADD.current_contact[0] = ADD.current_contact[0].replace("'", "")
        ADD.current_contact[0] = ADD.current_contact[0].replace(",", "")
        ADD.current_contact[1] = ADD.current_contact[1].replace("[", "")
        ADD.current_contact[1] = ADD.current_contact[1].replace("]", "")
        ADD.current_contact[1] = ADD.current_contact[1].replace("'", "")
        ADD.current_contact[1] = ADD.current_contact[1].replace(",", "")
        ADD.current_contact[2] = ADD.current_contact[2].replace("[", "")
        ADD.current_contact[2] = ADD.current_contact[2].replace("]", "")
        ADD.current_contact[2] = ADD.current_contact[2].replace("'", "")
        ADD.current_contact[2] = ADD.current_contact[2].replace(",", "")
        ADD.current_contact[3] = ADD.current_contact[3].replace("[", "")
        ADD.current_contact[3] = ADD.current_contact[3].replace("]", "")
        ADD.current_contact[3] = ADD.current_contact[3].replace("'", "")
        ADD.current_contact[3] = ADD.current_contact[3].replace(",", "")
        CONTACTS.contact += 1
    except IndexError:
        y = 10
        None


# Main menu loop
while True:
    # Clears the screen, prints the title and menu
    ADD.clear()
    print("C O N T A C T  O R G A N I Z E R\n")
    """f = open("SAVED_CONTACTS.txt","r+")
    f. truncate(0)"""
    print("""
What would you like to do?
\n[1] View current contacts
\n[2] Add new contact
\n[3] Save contacts and quit""")
    if example == True:
        print("\n[4] Add  example contacts because you have no friends")
    user_in = input(" ")
    ADD.clear()
    if user_in == "1":
        # Opens and displays contacts for the user
        CONTACTS.open_contact()
    elif user_in == "2":
        # Starts adding a contact
        MAIN = True
        ADD.ADD_CONTACT()
    elif user_in == "3":
        # Prints out current saved contacts
        while CONTACTS.contact >= 1:
            CONTACTS.contact -= 1
            ADD.CHECK_CONTACT()
            f = open("SAVED_CONTACTS.txt", "w")
            while CONTACTS.contact >= 1:
                ADD.CHECK_CONTACT()
                if ADD.current_contact[0] == "CONTACT DELETED":
                    ADD.current_contact.clear()
                    CONTACTS.contact -= 1
                    ADD.CHECK_CONTACT()
                else:
                    try:
                        f.write("\n" + ADD.current_contact[0])
                        f.write("\n" + ADD.current_contact[1])
                        f.write("\n" + ADD.current_contact[2])
                        f.write("\n" + ADD.current_contact[3])
                    except IndexError:
                        None
                CONTACTS.contact -= 1
        f = open("SAVED_CONTACTS.txt", "w")
        f.close()
        sys.exit()
    elif user_in == "4":
        # Adds a couple of contacts for ease of testing and showcase
        MAIN = True
        ADD.EXAMPLE_CONTACTS()
        example = False
